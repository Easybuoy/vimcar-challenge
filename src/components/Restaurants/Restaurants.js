import React from 'react';
import propTypes from 'prop-types';

const Restaurants = ({ businesses }) => {
  return (
    <>
      {businesses.map((business) => {
        return (
          <div className="card" key={business.id}>
            <img src={business.image_url} alt={business.name} />
            <div className="container">
              <h4>
                <a href={business.url}>{business.name}</a>
              </h4>
              {business.location && business.location.display_address && (
                <p>
                  {business.location.display_address[0]}
                  <br />
                  {business.location.display_address[1]}
                </p>
              )}
              <p>{business.display_phone}</p>
            </div>
          </div>
        );
      })}
    </>
  );
};

Restaurants.propTypes = {
    businesses: propTypes.array.isRequired,
}

Restaurants.displayName = 'components/Restaurants/Restaurants';

export default Restaurants;
