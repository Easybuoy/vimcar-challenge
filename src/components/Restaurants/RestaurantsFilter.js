import React from 'react';
import propTypes from 'prop-types';
import Select from 'react-select';

import FilterOptions from '../../constants/filterOptions';

const RestaurantsFilter = ({
  value,
  handleFilterChange,
  isLoading,
  options,
}) => {
  return (
    <div className="places-option-container">
      <Select
        className="places-option"
        value={value}
        onChange={handleFilterChange}
        options={options}
        isClearable
        isLoading={isLoading}
        placeholder="Choose restaurant"
      />
    </div>
  );
};

RestaurantsFilter.displayName = 'components/RestaurantsFilter';

RestaurantsFilter.defaultProps = {
  value: null,
  options: FilterOptions,
};

RestaurantsFilter.propTypes = {
  value: propTypes.shape({
    value: propTypes.string.isRequired,
    label: propTypes.string.isRequired,
  }),
  options: propTypes.arrayOf(
    propTypes.shape({
      value: propTypes.string.isRequired,
      label: propTypes.string.isRequired,
    })
  ).isRequired,
  handleFilterChange: propTypes.func.isRequired,
  isLoading: propTypes.bool.isRequired,
};

export default RestaurantsFilter;
