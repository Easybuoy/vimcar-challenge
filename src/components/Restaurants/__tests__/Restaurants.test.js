import React from 'react';
import { render } from '@testing-library/react';

import Restaurants from '../Restaurants';

describe('Restaurants', () => {
  it('should render and match styles', () => {
    const mockBusinesses = [
      {
        alias: 'stadtklause-berlin',
        categories: [
          {
            alias: 'german',
            title: 'German',
          },
        ],
        coordinates: { latitude: 52.504779, longitude: 13.380618 },
        display_phone: '+49 30 51056381',
        distance: 2205.244050414826,
        id: '8uBE9ZgfZOM2z_Rllw478Q',
        image_url:
          'https://s3-media3.fl.yelpcdn.com/bphoto/UtjeEmgk-RtAoTc7pJluDQ/o.jpg',
        is_closed: false,
        location: {
          address1: 'Bernburger Str. 35',
          address2: '',
          address3: null,
          city: 'Berlin',
          zip_code: '10963',
        },
        name: 'Stadtklause',
        phone: '+493051056381',
        price: 'â‚¬',
        rating: 4.5,
        review_count: 171,
        transactions: [],
        url: 'https://www.yelp.com/biz/stadtklause-berlin?adjust_creative=ors6zRuF6OV8AsDVv84wmA&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_search&utm_source=ors6zRuF6OV8AsDVv84wmA',
      },
    ];

    const { container } = render(
      <Restaurants businesses={mockBusinesses} />
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toMatchSnapshot();
  });
});
