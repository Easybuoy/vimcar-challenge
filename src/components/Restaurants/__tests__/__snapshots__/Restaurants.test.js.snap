// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restaurants should render and match styles 1`] = `
<div
  class="card"
>
  <img
    alt="Stadtklause"
    src="https://s3-media3.fl.yelpcdn.com/bphoto/UtjeEmgk-RtAoTc7pJluDQ/o.jpg"
  />
  <div
    class="container"
  >
    <h4>
      <a
        href="https://www.yelp.com/biz/stadtklause-berlin?adjust_creative=ors6zRuF6OV8AsDVv84wmA&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_search&utm_source=ors6zRuF6OV8AsDVv84wmA"
      >
        Stadtklause
      </a>
    </h4>
    <p>
      +49 30 51056381
    </p>
  </div>
</div>
`;

exports[`Restaurants should render and match styles 2`] = `
<div
  class="card"
>
  <img
    alt="Stadtklause"
    src="https://s3-media3.fl.yelpcdn.com/bphoto/UtjeEmgk-RtAoTc7pJluDQ/o.jpg"
  />
  <div
    class="container"
  >
    <h4>
      <a
        href="https://www.yelp.com/biz/stadtklause-berlin?adjust_creative=ors6zRuF6OV8AsDVv84wmA&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_search&utm_source=ors6zRuF6OV8AsDVv84wmA"
      >
        Stadtklause
      </a>
    </h4>
    <p>
      +49 30 51056381
    </p>
  </div>
</div>
`;
